### PR TITLE
MIGENG-333: MA - Store Account Number on Analysis_Model table

### PR DESCRIPTION
**Pull request by carlosthe19916**
https://issues.redhat.com/browse/MIGENG-333

Added `userAccountNumber` field to the AnalysisModel class

### CODE REVIEW

File: src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java [hunk 1]
```diff
@@ -317,6 +330,19 @@ public String getUserNameFromRHIdentity(String x_rh_identity_base64) {
         return result;
     }
 
+    public String getUserAccountFromRHIdentity(String x_rh_identity_base64) {
```
Comment (jonathanvila) [inline src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java]: As you are basically duplicating the code, probably you could refactor this method and mix it with the getUserNameFromRHIdentity in order to get both fields. Currently it is parsing the Json encoded value 2 times without need.

File: src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java [hunk 2]
```diff
@@ -222,6 +228,13 @@ public void configure() throws Exception {
                     exchange.getIn().setHeader(USERNAME, userName);
                 });
 
+        from("direct:add-userAccount-header")
```
Comment (jonathanvila) [inline src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java]: I haven't seen any test for this route similar to **MainRouteBuilder_DirectAddUsernameHeaderTest**

File: src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java (hunk 1 above)
Comment (carlosthe19916) [reply src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java]: Refactored

File: src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java (hunk 2 above)
Comment (carlosthe19916) [reply src/main/java/org/jboss/xavier/integrations/route/MainRouteBuilder.java]: removed route after refactoring

File: src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java [hunk 3]
```diff
@@ -103,7 +105,68 @@ public void mainRouteBuilder_routeDirectAddUsernameHeader_BadContentGiven_Should
...
 
         //Then
-        assertThat(result.getIn().getHeader("analysisUsername")).isNull();
+        assertThat(result.getIn().getHeader(RouteBuilderExceptionHandler.USERNAME)).isNull();
+        camelContext.stop();
+    }
+
+    @Test
+    public void mainRouteBuilder_xRhIdentityHeaderProcessor_givenInvalidHeader_shouldSetNullIdentityHeader() throws Exception {
+        // Given
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java]: if I dont understand wronly , you are testing a method *MainRouteBuilder.xRhIdentityHeaderProcessor* . So I dont see any point on creating a Camel context with only 1 route that executes that method.

Either you create a unit test to that method, or you test the real routes (starting only those affecting the new code).

File: src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java [hunk 4]
```diff
@@ -103,7 +105,68 @@ public void mainRouteBuilder_routeDirectAddUsernameHeader_BadContentGiven_Should
...
+
+        //Then
+        assertThat(result.getIn().getHeader(RouteBuilderExceptionHandler.USERNAME)).isNull();
+        camelContext.stop();
+    }
+
+    @Test
+    public void mainRouteBuilder_xRhIdentityHeaderProcessor_givenValidHeader_shouldSetUserIdentityHeaders() throws Exception {
+        // Given
+        camelContext.addRoutes(new RouteBuilder() {
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java]: if I dont understand wronly , you are testing a method **MainRouteBuilder.xRhIdentityHeaderProcessor** . So I dont see any point on creating a Camel context with only 1 route that executes that method.

Either you create a unit test to that method, or you test the real routes (starting only those affecting the new code).

File: src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectCalculateFlagSharedDisksTest.java [hunk 5]
```diff
@@ -33,6 +33,13 @@
     @Test
     public void mainRouteBuilder_DirectCalculate_JSONGiven_ShouldReturnExpectedCalculatedValues() throws Exception {
         //Given
+        camelContext.getRouteDefinition("flags-shared-disks").adviceWith(camelContext, new AdviceWithRouteBuilder() {
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectCalculateFlagSharedDisksTest.java]: What this change has to do with MIGENG-333 ?

File: src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectCalculateFlagSharedDisksTest.java [hunk 6]
```diff
@@ -79,6 +86,13 @@ public void mainRouteBuilder_DirectCalculate_JSONGiven_ShouldReturnExpectedCalcu
     @Test
     public void mainRouteBuilder_DirectCalculate_JSONOnVersion1_0_0Given_ShouldReturnExpectedCalculatedValues() throws Exception {
         //Given
+        camelContext.getRouteDefinition("flags-shared-disks").adviceWith(camelContext, new AdviceWithRouteBuilder() {
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectCalculateFlagSharedDisksTest.java]: What this change has to do with MIGENG-333 ?

File: src/test/java/org/jboss/xavier/utils/UtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
+package org.jboss.xavier.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
```
Comment (jonathanvila) [inline src/test/java/org/jboss/xavier/utils/UtilsTest.java]: Java6assertions ? this package is **deprecated**

Comment (carlosthe19916) [reply src/test/java/org/jboss/xavier/utils/UtilsTest.java]: changed for `import static org.assertj.core.api.Assertions.assertThat;`

File: src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java (hunk 3 above)
Comment (carlosthe19916) [reply src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java]: +1. I've removed the route that was created inside the test and use `camelContext.startRoute("add-username-header");` instead

File: src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java (hunk 4 above)
Comment (carlosthe19916) [reply src/test/java/org/jboss/xavier/integrations/route/MainRouteBuilder_DirectAddUsernameHeaderTest.java]: +1. I've removed the route that was created inside the test and use camelContext.startRoute("add-username-header"); instead

**Comment by carlosthe19916** [comment]
@jonathanvila I've fixed your comments

**Comment by jonathanvila** [comment]
Tested and OK : 

image

**Pull request merged**